app/src/main/java/com/wireguard/util/KeyedList.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright © 2017-2018 WireGuard LLC. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.wireguard.util
+
+/**
+ * A list containing elements that can be looked up by key. A `KeyedList` cannot contain
+ * `null` elements.
+ */
+
+interface KeyedList<K, E : Keyed<out K>> : java.util.List<E> {
```

Comment:
This is bad and nobody should be following suit. Use the Kotlin `List` class, because it's impossible your app has the spaghetti inheritance this garbage does.